hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/HealthTransformations.scala

```diff
@@ -0,0 +1,114 @@
+package org.broadinstitute.monster.dap
+
+import org.broadinstitute.monster.dogaging.jadeschema.table.HlesHealthCondition
+
+object HealthTransformations {
+
+  /** Parse all health-condition-related fields out of a raw RedCap record. */
+  def mapHealthConditions(rawRecord: RawRecord): Iterable[HlesHealthCondition] =
+    mapInfectiousDisease(rawRecord)
+
+  /** Generic helper method for creating Hles Health Condition rows. */
+  def createHealthConditionRow(
+    rawRecord: RawRecord,
+    conditionName: String,
+    conditionType: Long,
+    condition: Long,
+    isCongenital: Boolean = false,
+    conditionOtherDescription: Option[String] = None,
+    conditionCause: Option[Long] = None,
+    conditionCauseOtherDescription: Option[String] = None
+  ): Option[HlesHealthCondition] =
+    if (rawRecord.getBoolean(s"hs_${conditionName}")) {
+      Some(
+        HlesHealthCondition(
+          dogId = rawRecord.getRequired("study_id").toLong,
+          hsConditionType = conditionType,
+          hsCondition = condition,
+          hsConditionOtherDescription = conditionOtherDescription,
+          hsConditionIsCongenital = isCongenital,
+          hsConditionCause = conditionCause,
+          hsConditionCauseOtherDescription = conditionCauseOtherDescription,
+          hsDiagnosisYear = rawRecord.getOptionalNumber(s"hs_${conditionName}_year"),
+          hsDiagnosisMonth = rawRecord.getOptionalNumber(s"hs_${conditionName}_month"),
+          hsRequiredSurgeryOrHospitalization =
+            rawRecord.getOptionalNumber(s"hs_${conditionName}_surg"),
+          hsFollowUpOngoing = rawRecord.getOptionalBoolean(s"hs_${conditionName}_fu")
+        )
+      )
+    } else {
+      None
+    }
+
+  /** Parse all infectious disease related fields out of a raw RedCap record. */
+  def mapInfectiousDisease(rawRecord: RawRecord): Iterable[HlesHealthCondition] =
+    if (rawRecord.getBoolean("hs_dx_infectious_yn")) {
+      // iterate over all infectious diseases
+      infectious_diseases.flatMap {
+        // "other" case
+        case ("infect_other", categorical) =>
+          createHealthConditionRow(
+            rawRecord,
+            s"dx_infect_other",
+            1L,
+            categorical.toLong,
```

Comment:
Would be better to have the map values be `Long` to start. The compiler might complain if you don't add the `L` suffix to each number.